src/Core/Services/Implementations/PolicyService.cs

```diff
@@ -57,6 +60,12 @@ public class PolicyService : IPolicyService
                         {
                             throw new BadRequestException("Single Sign-On Authentication policy is enabled.");
                         }
+
+                        var ssoConfig = await _ssoConfigRepository.GetByOrganizationIdAsync(org.Id);
+                        if (ssoConfig?.GetData()?.UseKeyConnector == true)
+                        {
+                            throw new BadRequestException("KeyConnector is enabled.");
+                        }
```

Comment:
Lets put this in a private method to avoid adding more logic to this function. We could do this for all SinlgeOrg policies if desired, but not needed if that is considered out of scope.

Reply:
I think we need to look into refactoring the whole service rather than moving things to private methods. Ideally each policy should have it's own class with it's own validate method instead of having a switch case.

Reply:
I'm okay with leaving this as is if we are recording the tech debt somewhere that it will be prioritized, same with the UserService comments.

Reply:
Added a tech debt ticket.